sdk/resourcemanager/Azure.ResourceManager/README.md

```diff
@@ -92,8 +85,8 @@ Console.WriteLine($"ResourceGroup: {id.ResourceGroupName}");
 Console.WriteLine($"Vnet: {id.Parent.Name}");
 Console.WriteLine($"Subnet: {id.Name}");
 ```
-
-#### Casting to the base resource identifier
+However, keep in mind that some of those properties could be null. You can usually tell by the id string itself which type a resource ID is, but if you are unsure, use the Try methods to retrieve the values as it's shown below:
```

Comment:
I would say you can check if the properties are null OR use the try methods

again use the ctor for ResourceIdentifier